pkg/traceutil/trace.go

```diff
@@ -102,13 +104,26 @@ func (t *Trace) InsertStep(at int, time time.Time, msg string, fields ...Field)
 	}
 }
 
+// SubTrace adds step to trace as a sublevel trace
+func (t *Trace) SubTrace(fields ...Field) {
+	t.steps = append(t.steps, step{fields: fields, isSubTrace: true})
+}
+
 // Step adds step to trace
 func (t *Trace) Step(msg string, fields ...Field) {
 	if !t.stepDisabled {
 		t.steps = append(t.steps, step{time: time.Now(), msg: msg, fields: fields})
 	}
 }
 
+// StepWithFunction will measure the input function as a single step
+func (t *Trace) StepWithFunction(f func(), msg string, fields ...Field) {
```

Comment:
I like the idea of `StepWithFunction`, do we still need to expose `EnableStep` and `DisableStep`?

Reply:
Yes you are right. I changed them to private functions.